target_snowflake/__init__.py

```diff
@@ -257,6 +276,26 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
                 else:
                     stream_to_sync[stream] = DbSync(config, o, table_cache, file_format_type)
 
+                if archive_load_files_enabled:
+                    # Determine archive_load_files_primary_column
+                    # 1) Use incremental replication key if defined
+                    # 2) Otherwise use primary key
+                    if False and 'bookmark_properties' in o and len(o['bookmark_properties']) > 0:
+                        LOGGER.info("Using %s from bookmark_properties as archive_load_files_primary_column",
+                                    o['bookmark_properties'][0])
+                        archive_load_files_primary_column = o['bookmark_properties'][0]  # TBD Can there be more than 1?
+                    else:
+                        LOGGER.info("Using %s from key_properties as archive_load_files_primary_column",
+                                    o['key_properties'][0])
+                        archive_load_files_primary_column = o['key_properties'][0]  # Behavior when key multi-col?
+
+                    archive_load_files_data[stream] = {
+                        'tap': config.get('id'),
```

Comment:
```suggestion
                        'tap': config.get('tap_id'),
```

target-snowflake can't access the original YAML file. The config is a generated JSON of the items under `db_conn` and some other propertes listed [here](https://github.com/transferwise/pipelinewise/blob/eca52d4fa566f27c53b206a59ada86ee7d986a68/pipelinewise/cli/config.py#L320)